pytest_bdd/gherkin_terminal_reporter.py

```diff
@@ -0,0 +1,99 @@
+# -*- encoding: utf-8 -*-
+import py
+import pytest
+
+from _pytest.terminal import TerminalReporter
+
+
+def add_options(parser):
+    group = parser.getgroup("terminal reporting", "reporting", after="general")
+    group._addoption(
+        '--gherkin-terminal-reporter',
+        action="store_true",
+        dest="gherkin_terminal_reporter",
+        default=False,
+        help=(
+            "enable gherkin output"
+        )
+    )
+
+
+@pytest.mark.trylast
+def pytest_configure(config):
```

Comment:
There's only one pytest_configure per plugin. The previous one is implemented in the cucumber_json.py and imported to plugis as import *.
You did the same, but for the terminal reporter. Import * has overwritten the cucumber_json one so that doesn't exist anymore within plugin.py

I think we need to implement one pytest_configure inside the plugin.py and call a function from each of cucumber_json and gherkin_terminal_reporter to register those.

Same for unconfigure. I'm afraid you have to do a little refactoring there. Also import just a module to call cucumber_json.configure/unconfigure instead of *.